crates/text_edit/src/lib.rs

```diff
@@ -175,6 +175,12 @@ impl TextEditBuilder {
     pub fn invalidates_offset(&self, offset: TextSize) -> bool {
         self.indels.iter().any(|indel| indel.delete.contains_inclusive(offset))
     }
+    fn indel(&mut self, indel: Indel) {
+        self.indels.push(indel);
+        if self.indels.len() <= 16 {
+            check_disjoint(&mut self.indels);
```

Comment:
Should this be something like `assert!(check_disjoint(&mut self.indels))`?

Reply:
Excellent catch @samlh! Thanks!